### ISSUE TITLE
panicked at 'range end index 40 out of range for slice of length 32'

### ISSUE THREAD

**Issue by rcgoodfellow** [oxidecomputer/opte#355]
My machine had been running an Omicron-launched instance for about a day with little to no activity going on inside the VM. The host machine crashed with the follwing.

```
> ::status
debugging crash dump vmcore.0 (64-bit) from masaka
operating system: 5.11 helios-1.0.21472 (i86pc)
build version: heads/master-0-g8b0ccc49c4

image uuid: c166060a-e746-cacc-b09d-c4cfa7aa34de
panic message: panicked at 'range end index 40 out of range for slice of length 32', /work/oxidecomputer/opte/opte/src/engine/tcp.rs:128:17
dump content: kernel pages only
```
```
> $G
C++ symbol demangling enabled
> $C
fffffcc271fc65f0 vpanic()
fffffcc271fc6680 rust_begin_unwind+0x146()
fffffcc271fc66c0 core::panicking::panic_fmt::hbd42276bcaa2dfd8+0x2d()
fffffcc271fc6730 core::slice::index::slice_end_index_len_fail_rt::hf844c6accdd72c4f+0x64()
fffffcc271fc6740 core::slice::index::slice_end_index_len_fail::hc2c5759eaf42ff3b+9()
fffffcc271fc6840 opte::engine::packet::Packet<opte::engine::packet::Initialized>::parse_tcp::h6644c11f5a04a7bf+0x54a()
fffffcc271fc6cd0 <oxide_vpc::engine::VpcParser as opte::engine::NetworkParser>::parse_outbound::he0b7121607667c75+0x9d7()
fffffcc271fc7330 opte::engine::packet::Packet<opte::engine::packet::Initialized>::parse::hb612752e024c5dd6+0x145()
fffffcc271fc7c40 xde_mc_tx+0xfb()
fffffcc271fc7c70 mac_ring_tx+0x43(fffffe2aefb5d030, 0, fffffe2de7aaa620)
fffffcc271fc7cc0 mac_provider_tx+0x85(fffffe2aefb5d030, 0, fffffe2de7aaa620, fffffe2a800dad68)
fffffcc271fc7d80 mac_tx+0x295(fffffe2a800dad68, fffffe2de7aaa620, 0, 1, 0)
fffffcc271fc7e40 viona_tx+0x48c(fffffe2398876400, fffffe2398876540)
fffffcc271fc7ea0 viona_worker_tx+0x73(fffffe2398876540, fffffe2398876400)
fffffcc271fc7ef0 viona_worker+0x126(fffffe2398876540)
fffffcc271fc7f00 thread_start+0xb()
```

**Comment by rcgoodfellow** [comment]
I have a core dump for this, but the file is rather large. Can provide on request.

**Comment by bnaecker** [comment]
Looks similar to https://github.com/oxidecomputer/opte/issues/342.

**Comment by rcgoodfellow** [comment]
Yup sure does. I was only looking at open tickets 🤦. Will check what version of OPTE I have.

**Comment by rcgoodfellow** [comment]
Ok - I'm running version OPTE 0.21.158 and it looks like this was fixed in 0.21.160.

Closing.

Thanks for the pointer @bnaecker.
